File: watchout/Demo1/src/robot_decision/scripts/talker.py
#!/usr/bin/env python
import rospy
from std_msgs.msg import String

def main():
    rospy.init_node("talker")
    pub1 = rospy.Publisher("/pub1",String,queue_size=10)
    pub2 = rospy.Publisher("/pub2",String,queue_size=10)
    rate = rospy.Rate(10)
    while not rospy.is_shutdown():
        helloworld1 = "hellow_world_1 %s" % rospy.get_rostime()
        helloworld2 = "hellow_world_2 %s" % rospy.get_rostime()

        print(helloworld1)
        pub1.publish(helloworld1)
        pub2.publish(helloworld2)
        rate.sleep()

if __name__=="__main__":
    try:
        main()
    except rospy.ROSException:
        pass

File: watchout/Demo1/src/robot_perception/src/talker.cpp
#include<ros/ros.h>
#include<std_msgs/String.h>
#include<string>

using namespace std;

int main(int argc,char ** argv){
    ros::init(argc,argv,"talker");
    ros::NodeHandle n1("n1");
    ros::NodeHandle n2("n2");
    ros::Publisher pub1 = n1.advertise<std_msgs::String>("/helloworld1",10);
    ros::Publisher pub2 = n2.advertise<std_msgs::String>("/helloworld2",10);
    ros::Rate rate(10);
    int i = 0;
    while (ros::ok())
    {
        std_msgs::String msg1,msg2;

        std::stringstream s1,s2;
        s1 << "hello world " << ros::Time::now();
        msg1.data = s1.str();
        s2 << "hello world " << "2";
        // msg2.data = s2.str(  
        pub1.publish(msg1);
        pub2.publish(msg2);
        std::cout<< msg1.data.c_str()<<std::endl;
        std::cout<< msg2.data.c_str()<<std::endl;
        rate.sleep();
        i++;
    }
    
}
